test/LCS.test.js

```diff
@@ -31,5 +31,13 @@ test('LCS', t => {
     t.end();
   });
 
+  t.test('with timeout', t => {
```

Comment:
this test could falsely pass / fail depending on the hardware it runs on
this is not a good thing to rely on in unit tests

maybe the test that is expected to fail should have a 0 timeout?
either that, or mock `Date.now`

the same applies to all tests